src/rgw/rgw_op.cc

```diff
@@ -3943,6 +3943,8 @@ void RGWPutObj::execute(optional_yield y)
 
   rgw_placement_rule *pdest_placement = &s->dest_placement;
 
+  s->object->set_trace(s->trace->GetContext());
```

Comment:
what if `s->trace` is null?

Reply:
this scenario is not possible. `s->trace` is initialized unconditionally before the op execution(in rgw_process.cc)

in case we compile with `WITH_JAEGER=OFF` then `s->trace` won't be a ptr, but a stub struct (with `->` overloading)